### PR TITLE
OSD-6853: CatalogSource by digest

### PR DESCRIPTION
**Pull request by 2uasimojo**
[OSD-6853](https://issues.redhat.com/browse/OSD-6853)

### CODE REVIEW

**Comment by georgettica** [comment]
Hey, the REPO_DIGEST is not in the template file and thus won't work when replacing.

There is a process that deploys RMO using the template (so you can test this works)

Would you want me to outline it in a different PR so you can follow?
If not would you like me to test it to verify it's working as expected?

**Comment by 2uasimojo** [comment]
/hold

Whoah, one of my deltas didn't make it. Weird.

**Comment by 2uasimojo** [comment]
> make generate-syncset

[This automated commit](https://github.com/openshift/route-monitor-operator/pull/107/commits/fa50b8aefc44aa29e035ad1fcb1aa3ee979f7107) removed the required var from the template. WTF?

**Comment by 2uasimojo** [comment]
> Hey, the REPO_DIGEST is not in the template file and thus won't work when replacing.
> 
> There is a process that deploys RMO using the template (so you can test this works)
> 
> Would you want me to outline it in a different PR so you can follow?
> If not would you like me to test it to verify it's working as expected?

I think I figured it out @georgettica -- can you please confirm on the latest rev?

File: hack/olm-registry/olm-artifacts-template.yaml [hunk 1]
```diff
@@ -5,6 +5,8 @@ parameters:
   required: true
 - name: IMAGE_TAG
   required: true
+- name: REPO_DIGEST
+  required: true
```
Comment (2uasimojo) [inline hack/olm-registry/olm-artifacts-template.yaml:9]: ✓

**Review by 2uasimojo — COMMENTED** [review]
/hold cancel

This is better.

**Comment by georgettica** [comment]
> > Hey, the REPO_DIGEST is not in the template file and thus won't work when replacing.
> > There is a process that deploys RMO using the template (so you can test this works)
> > Would you want me to outline it in a different PR so you can follow?
> > If not would you like me to test it to verify it's working as expected?
> 
> I think I figured it out @georgettica -- can you please confirm on the latest rev?

didn't see it

more active on other channels

**Pull request merged**